src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -17,14 +19,19 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
-
+        requireNonNull(args);
         if (args.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
         }
 
-        Name name = ParserUtil.parseName(args);
-        return new DeleteCommand(name);
+        if (args.trim().matches("\\d+")) {
+            Index index = ParserUtil.parseIndex(args);
+            return new DeleteCommand(index);
+        } else {
+            Name name = ParserUtil.parseName(args);
+            return new DeleteCommand(name);
+        }
```

Comment:
Good job in transisting the recent version for command parser test to a more relevant version.